src/OrchardCore.Modules/OrchardCore.Menu/AdminMenu.cs

```diff
@@ -24,6 +24,7 @@ public Task BuildNavigationAsync(string name, NavigationBuilder builder)
 
             var rvd = new RouteValueDictionary
             {
+                { "contentTypeId", "Menu" },
```

Comment:
By my test, I pretty sure that is necessary because when you click over a link of a specific contenttype the "Content Item" link change and take the contentTypeId of the "last link" clicked: where "last link" is a link that list the contentitem of a specific contenttype